New `ubuntu-latest` image prevents isort from being installed

Apparently GitHub actions have changed their `ubuntu-latest` runner.   In the current runner, a python virtual environment is required to install python packages.

```console
Run isort/isort-action@v1
Run /home/runner/work/_actions/isort/isort-action/v1/bin/ensure_python
Run /home/runner/work/_actions/isort/isort-action/v1/bin/install_packages latest requirements.txt
Install isort
  Installing latest version of isort
  error: externally-managed-environment
  
  × This environment is externally managed
  ╰─> To install Python packages system-wide, try apt install
      python3-xyz, where xyz is the package you are trying to
      install.
      
      If you wish to install a non-Debian-packaged Python package,
      create a virtual environment using python3 -m venv path/to/venv.
      Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
      sure you have python3-full installed.
      
      If you wish to install a non-Debian packaged Python application,
      it may be easiest to use pipx install xyz, which will manage a
      virtual environment for you. Make sure you have pipx installed.
      
      See /usr/share/doc/python3.12/README.venv for more information.
  
  note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
  hint: See PEP 668 for the detailed specification.
Install modules from requirements arg
Run /home/runner/work/_actions/isort/isort-action/v1/bin/run_isort --check-only --diff examples src/template_python tests
Running isort --check-only --diff examples src/template_python tests
/home/runner/work/_actions/isort/isort-action/v1/bin/run_isort: line 4: isort: command not found
Error: Process completed with exit code 127.
```

So something like the following is needed:

```console
          python -m venv ~/venv
          source ~/venv/bin/activate
          python -m pip install --upgrade pip
          python -m pip install --requirement requirements.txt
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - A [problem matcher](https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md) so you can see issues without trawling through logs
 
+### Fixed
+
+- Packages are installed into a virtualenv due to the fact that system-wide
+  installs fail by default when run on newer versions of Python.
+
 ## [1.1.0] - 2022-11-04
 
 - All inputs changed to use kebab-case: `isort-version`, `sort-paths`, and

--- a/action.yml
+++ b/action.yml
@@ -73,6 +73,9 @@ runs:
     - run: $GITHUB_ACTION_PATH/bin/ensure_python
       shell: bash
     - run: >
+        python -m venv "${HOME}/isort-venv"
+      shell: bash
+    - run: >
         $GITHUB_ACTION_PATH/bin/install_packages
         ${{ inputs.isort-version || inputs.isortVersion }}
         ${{ inputs.requirements-files || inputs.requirementsFiles }}

--- a/bin/install_packages
+++ b/bin/install_packages
@@ -4,6 +4,10 @@ isort_version=$1
 requirements_files=$2
 
 echo "::group::Install isort"
+
+# shellcheck source=/dev/null
+source "${HOME}/isort-venv/bin/activate"
+
 if [ -z "$isort_version" ] || [ "$isort_version" == "latest" ]; then
     echo "Installing latest version of isort"
     python3 -m pip install "isort[requirements_deprecated_finder,pipfile_deprecated_finder]"

--- a/bin/run_isort
+++ b/bin/run_isort
@@ -1,5 +1,8 @@
 #!/usr/bin/env bash
 
+# shellcheck source=/dev/null
+source "${HOME}/isort-venv/bin/activate"
+
 echo "Running isort $*"
 isort_result=$(isort "$@")
 exit_code=$?

